src/otx/algo/classification/huggingface_model.py

```diff
@@ -13,25 +13,24 @@
 
 from otx.core.data.entity.base import OTXBatchLossEntity
 from otx.core.data.entity.classification import (
-    MulticlassClsBatchDataEntity,
+    CLASSIFICATION_BATCH_DATA_ENTITY,
     MulticlassClsBatchPredEntity,
 )
-from otx.core.exporter.base import OTXModelExporter
-from otx.core.exporter.native import OTXNativeModelExporter
-from otx.core.metrics.accuracy import MultiClassClsMetricCallable
+from otx.core.metrics.accuracy import DefaultClsMetricCallable
 from otx.core.model.base import DefaultOptimizerCallable, DefaultSchedulerCallable
-from otx.core.model.classification import OTXMulticlassClsModel
+from otx.core.model.classification import OTXClassificationModel
 from otx.core.schedulers import LRSchedulerListCallable
 from otx.core.types.label import LabelInfoTypes
+from otx.core.types.task import OTXTaskType, OTXTrainType
 
 if TYPE_CHECKING:
     from lightning.pytorch.cli import LRSchedulerCallable, OptimizerCallable
     from transformers.modeling_outputs import ImageClassifierOutput
 
-    from otx.core.metrics import MetricCallable
+    from otx.core.metrics import MetricCallablePerTask
 
 
-class HuggingFaceModelForMulticlassCls(OTXMulticlassClsModel):
+class HuggingFaceModelForMulticlassCls(OTXClassificationModel):
```

Comment:
Just a question, is the reason why `MulticlassCls` is in the class name that huggingface only supports multi class cls?

Reply:
I'm thinking about removing classes like `OTXMulticlassClsModel` (sub-task OTXModel) in the next PR, so now i'm considering about unifying the `OTXModel` used by all classification models into an `OTXClassificationModel` for now. Any other thoughts on this?